### PR TITLE
Bump to Bazel 3.6.0

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Bumps to bazel-toolchains for Bazel 3.6.0 and regenerates toolchains for Bazel 3.6.0

### CODE REVIEW

**Comment by lizan** [comment]
I'm late to this, let's bump to 3.6.0 ?

File: WORKSPACE [hunk 1]
```diff
@@ -4,9 +4,9 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
 http_archive(
     name = "bazel_toolchains",
-    sha256 = "09642745581d071c7cd670e57589ea1ba78c99e9d7edb8b7478bda2640773e90",
-    strip_prefix = "bazel-toolchains-022dcbb331a80548cefecc289f3741b98b882307",
-    urls = ["https://github.com/bazelbuild/bazel-toolchains/archive/022dcbb331a80548cefecc289f3741b98b882307.tar.gz"],
+    sha256 = "89a053218639b1c5e3589a859bb310e0a402dedbe4ee369560e66026ae5ef1f2",
```
Comment (moderation) [inline WORKSPACE:7]: `sha256 = "4fb3ceea08101ec41208e3df9e56ec72b69f3d11c56629d6477c0ff88d711cf7"`

File: WORKSPACE [hunk 2]
```diff
@@ -4,9 +4,9 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
...
+    strip_prefix = "bazel-toolchains-3.5.0",
```
Comment (moderation) [inline WORKSPACE:8]: `strip_prefix = "bazel-toolchains-3.6.0",`

File: WORKSPACE [hunk 3]
```diff
@@ -4,9 +4,9 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
...
+    urls = ["https://github.com/bazelbuild/bazel-toolchains/releases/download/3.5.0/bazel-toolchains-3.5.0.tar.gz"],
```
Comment (moderation) [inline WORKSPACE:9]: `urls = ["https://github.com/bazelbuild/bazel-toolchains/releases/download/3.6.0/bazel-toolchains-3.6.0.tar.gz"],`

File: toolchains/regenerate.sh [hunk 4]
```diff
@@ -38,7 +38,7 @@ esac
 
 # Bazel query is the right command so bazel won't fail itself.
 # Keep bazel versions here at most two: current master version, next version
-for BAZEL_VERSION in "3.3.1" "3.4.1"; do
+for BAZEL_VERSION in "3.4.1" "3.5.0"; do
```
Comment (moderation) [inline toolchains/regenerate.sh:41]: for BAZEL_VERSION in "3.5.1" "3.6.0"; do

**Comment by sunjayBhatia** [comment]
@lizan @moderation is there a good pattern you have used before in AZP with a periodic pipeline or something we can use automate this?

**Comment by moderation** [comment]
@sunjayBhatia We haven't put in place automatic dependency updates yet but I think this is something @htuch might be open too for non-controlplane, non-dataplane-* dependencies. I notice that https://dependabot.com/ is in use in https://github.com/envoyproxy/envoy-tools/actions but I'm not sure how or if that could work monitoring for Github releases of commits - https://dependabot.com/#languages

File: toolchains/regenerate.sh [hunk 5]
```diff
@@ -38,7 +38,7 @@ esac
 
 # Bazel query is the right command so bazel won't fail itself.
 # Keep bazel versions here at most two: current master version, next version
-for BAZEL_VERSION in "3.3.1" "3.4.1"; do
+for BAZEL_VERSION in "3.5.1" "3.6.0"; do
```
Comment (lizan) [inline toolchains/regenerate.sh:41]: Let's keep the first as `3.4.1` so it won't block us updating build-tools sha in envoyproxy/envoy.

**Review by lizan — APPROVED** [review]

**Pull request merged**